MetaMask/metamask-mobile
# fix: JS "pseudo protocol" works when pasted on mobile browser

## **Description**

Updated originWhitelist and blocked javascript entries in the onSubmitUrl.

This fix stops users being able to inject Javascript scripts into the in app web browser.

## **Related issues**

Fixes: https://github.com/MetaMask/pm-security/issues/258

## **Manual testing steps**

1. Open the in app browser and go to [w2d2rd.csb.app](https://w2d2rd.csb.app/)
2. Follow the onscreen instructions and copy paste the link into the address bar
3. Observe that the alert does not fire and instead the url is undefined.

## **Screenshots/Recordings**

### **Before**
https://github.com/MetaMask/metamask-mobile/assets/34304333/651e69ad-7d95-4e73-9173-bbbbf414388a

### **After**
https://github.com/MetaMask/metamask-mobile/assets/34304333/d1275f29-3235-4e47-8b50-b50527f5a41e

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
